tulip/synth.py

```diff
@@ -980,28 +980,42 @@ def synthesize_many(specs, ts=None, ignore_init=None,
     assert isinstance(ts, dict)
     for name, t in ts.items():
         assert isinstance(t, transys.FiniteTransitionSystem)
-        ignore = name in ignore_init
-        bool_act = name in bool_actions
+        ignore = ignore_init.get(name) 
+        bool_act = bool_actions.get(name)
         statevar = name
         if t.owner == 'sys':
-            specs |= sys_to_spec(t, ignore, statevar,
+            sys_spec = sys_to_spec(t, ignore, statevar,
                                  bool_actions=bool_act)
+            sys_spec.qinit = specs.qinit 
+            sys_spec.moore = specs.moore 
+            sys_spec.plus_one = specs.plus_one 
+            specs |= sys_spec
```

Comment:
This is was an error, thanks for addressing it!

Previous behavior: even if all `ts.values()` had the same `qinit, moore, plus_one` attributes, the function `sys_to_spec` ignored them, returning a `GRSpec` with the defaults:

https://github.com/tulip-control/tulip-control/blob/1ad5b12ce1f96253d898abb8485d4e0ca8d20d90/tulip/synth.py#L493

This could introduce a mismatch between specs when `__or__` is applied:

https://github.com/tulip-control/tulip-control/blob/1ad5b12ce1f96253d898abb8485d4e0ca8d20d90/tulip/spec/form.py#L613-L618

This error didn't arise in the function `synthesize` because it copies the values of those attributes after calling `sys_to_spec`:

https://github.com/tulip-control/tulip-control/blob/1ad5b12ce1f96253d898abb8485d4e0ca8d20d90/tulip/synth.py#L1215-L1218

It seems that this change can move to within `sys_to_spec`, thus address all call sites.

Reply:
However, attribute values should be copied from `t`, whenever defined there, and only otherwise from `specs`.

Regarding moving this change to the function `sys_to_spec` (and `env_to_spec`): such a refactoring would require propagating `specs` (for the default attribute values) to `sys_to_spec` (and `env_to_spec`). This isn't convenient.